src/extremeweatherbench/utils.py

```diff
@@ -558,3 +558,24 @@ def derive_indices_from_init_time_and_lead_time(
     init_time_subset_indices = valid_time_indices[0]
 
     return init_time_subset_indices
+
+
+def maybe_convert_dataset_lead_time_to_int(dataset: xr.Dataset) -> xr.Dataset:
+    """Convert types of variables in an xarray Dataset based on the schema,
+    ensuring that, for example, the variable representing lead_time is of type int.
```

Comment:
nit: Prefix method name with "_" (`_maybe_convert_dataset_lead_time_to_int`) to emphasize it's a private method and users should not attempt to use it unless they know what they're doing.